flash_attn/models/llama.py

```diff
@@ -308,7 +309,28 @@ def config_from_meta_checkpoint(
         num_attention_heads=params["n_heads"],
         num_hidden_layers=params["n_layers"],
         rms_norm_eps=params["norm_eps"],
+        num_key_value_heads=params.get("n_kv_heads", None),
     )
+    multiple_of = params.get("multiple_of", 1)
+    ffn_dim_multiplier = params.get("ffn_dim_multiplier", None)
+
+    intermediate_size = 4 * config.hidden_size
+    intermediate_size = int(2 * intermediate_size / 3)
+    # custom dim factor multiplier
+    if ffn_dim_multiplier is not None:
+        intermediate_size = int(ffn_dim_multiplier * intermediate_size)
+    intermediate_size = multiple_of * ((intermediate_size + multiple_of - 1) // multiple_of)
+    config.intermediate_size = intermediate_size
+    if "rope_theta" in params:
+        config.rotary_emb_base = params["rope_theta"]
+    config.vocab_size = 32000
```

Comment:
being slightly pedantic here, but extracting the `vocab_size` based on the `model_name` can feel slightly ad-hoc. can we unify the two settings by always loading the `tokenizer.model` file and measuring the size of the resulting tokenizer? 

e.g., i might want to load another fine-tuned model from codellama, which has modelname `custom-my-model`, which doesn't contain the unigram `code` in the name.

Reply:
Good point - I've changed the code. Note that the regular LLaMa models don't contain the tokenizer within the checkpoint folder, so we will naturally fallback to 32k which is correct.